packages/eth/src/eip_4844/types.rs

```diff
@@ -0,0 +1,266 @@
+use std::{ffi::CString, ops::Deref};
+
+use ethers::{
+    core::k256::sha2::{Digest, Sha256},
+    types::{Address, Signature, H256, U256},
+    utils::keccak256,
+};
+use lazy_static::lazy_static;
+use rlp::RlpStream;
+
+lazy_static! {
+    static ref KZG_SETTINGS: c_kzg::KzgSettings = c_kzg::KzgSettings::load_trusted_setup_file(
+        &CString::new("packages/eth/src/eip_4844/trusted_setup.txt").expect("C string"),
+    )
+    .unwrap();
+}
+
+const BLOB_TX_TYPE: u8 = 0x03;
+const VERSIONED_HASH_VERSION_KZG: u8 = 1;
+const MAX_BLOBS_PER_BLOCK: usize = 6;
+pub const MAX_BYTES_PER_BLOB: usize = c_kzg::BYTES_PER_BLOB;
+
+pub trait BlobSigner {
+    fn sign_hash(&self, hash: H256) -> std::result::Result<Signature, String>;
+}
+
+impl BlobSigner for ethers::signers::LocalWallet {
+    fn sign_hash(&self, hash: H256) -> std::result::Result<Signature, String> {
+        self.sign_hash(hash).map_err(|e| e.to_string())
+    }
+}
+
+pub struct PreparedBlob {
+    pub commitment: Vec<u8>,
+    pub proof: Vec<u8>,
+    pub versioned_hash: H256,
+    pub data: Vec<u8>,
+}
+
+pub struct BlobSidecar {
+    blobs: Vec<PreparedBlob>,
+}
+
+impl BlobSidecar {
+    pub fn new(data: Vec<u8>) -> std::result::Result<Self, String> {
+        let num_blobs = data.len().div_ceil(MAX_BYTES_PER_BLOB);
+
+        if num_blobs > MAX_BLOBS_PER_BLOCK {
+            return Err(format!(
+                "Data cannot fit into the maximum number of blobs per block: {}",
+                MAX_BLOBS_PER_BLOCK
+            ));
+        }
+
+        let field_elements = Self::partition_data(data);
+        let blobs = Self::field_elements_to_blobs(field_elements);
+        let prepared_blobs = blobs.iter().map(Self::prepare_blob).collect();
+
+        Ok(Self {
+            blobs: prepared_blobs,
+        })
+    }
+
+    pub fn num_blobs(&self) -> usize {
+        self.blobs.len()
+    }
+
+    pub fn versioned_hashes(&self) -> Vec<H256> {
+        self.blobs.iter().map(|blob| blob.versioned_hash).collect()
+    }
+
+    fn partition_data(data: Vec<u8>) -> Vec<[u8; 32]> {
```

Comment:
I've added a comment to provide some more context, combined with the EIP-4844 specs, it should bring a future maintainer closer to a full picture. 

In the near future we should move from ethers to alloy. This would also alleviate maintenance costs since alloy provides support for blob transactions.

Same applies to similar questions below.